### ISSUE TITLE
FR: Would like support for setting board orientation

### ISSUE THREAD

**Issue by davidair** [vpatkov/fen2png#1]
Currently the board is always rendered from White's standpoint (white at the bottom). Can you add support for setting board orientation either automatically (based on whose turn it is in the FEN, w or b) or manually via a flag?

Here is a possible reference implementation (for autodetect):

```
diff --git a/fen2png.go b/fen2png.go
index 38583f3..3275ae2 100644
--- a/fen2png.go
+++ b/fen2png.go
@@ -83,6 +83,48 @@ func check(err error) {
 	}
 }
 
+func getBoard(ranks []string, f *chessFont, flip bool) (rows [8][8]rune, err error) {
+	for y, rank := range ranks {
+		x := 0
+		for _, piece := range rank {
+
+			outputColum := x
+			outputRow := y
+			if flip {
+				outputColum = 7 - outputColum
+				outputRow = 7 - outputRow
+			}
+
+			if piece >= '1' && piece <= '8' {
+				r := f.pieces[' ']
+				for i := 0; i < int(piece)-int('0'); i++ {
+					rows[outputRow][outputColum] = r[(x+y)%2]
+					x++
+					outputColum = x
+					if flip {
+						outputColum = 7 - outputColum
+					}
+				}
+			} else {
+				r, ok := f.pieces[piece]
+				if ok {
+					rows[outputRow][outputColum] = r[(x+y)%2]
+					x++
+				} else {
+					return rows, fmt.Errorf("unknown piece %q in FEN", piece)
+				}
+			}
+		}
+
+		if x != 8 {
+			return rows, fmt.Errorf("%d files in FEN at rank %q", x, rank)
+		}
+
+	}
+
+	return rows, nil
+}
+
 func decodeFEN(fen string, f *chessFont) (rows []string, err error) {
 	fields := strings.Fields(fen)
 	if len(fields) == 0 {
@@ -92,6 +134,10 @@ func decodeFEN(fen string, f *chessFont) (rows []string, err error) {
 	if len(ranks) != 8 {
 		return nil, fmt.Errorf("%d ranks in FEN", len(ranks))
 	}
+	flip := false
+	if len(fields) > 1 && fields[1] == "b" {
+		flip = true
+	}
 
 	// Top
 	var row strings.Builder
@@ -102,30 +148,15 @@ func decodeFEN(fen string, f *chessFont) (rows []string, err error) {
 	row.WriteRune(f.topRightCorner)
 	rows = append(rows, row.String())
 
+	board, err := getBoard(ranks, f, flip)
+	check(err)
+
 	// Middle
-	for y, rank := range ranks {
+	for _, boardRow := range board {
 		row.Reset()
 		row.WriteRune(f.leftSide)
-		x := 0
-		for _, piece := range rank {
-			if piece >= '1' && piece <= '8' {
-				r := f.pieces[' ']
-				for i := 0; i < int(piece)-int('0'); i++ {
-					row.WriteRune(r[(x+y)%2])
-					x++
-				}
-			} else {
-				r, ok := f.pieces[piece]
-				if ok {
-					row.WriteRune(r[(x+y)%2])
-					x++
-				} else {
-					return nil, fmt.Errorf("unknown piece %q in FEN", piece)
-				}
-			}
-		}
-		if x != 8 {
-			return nil, fmt.Errorf("%d files in FEN at rank %q", x, rank)
+		for _, boardRune := range boardRow {
+			row.WriteRune(boardRune)
 		}
 		row.WriteRune(f.rightSide)
 		rows = append(rows, row.String())
```

**Comment by vpatkov** [comment]
Initially I also implemented flip and auto-flip features, but then I deliberately removed them, because flipped diagrams actually only leads to confusion. Coordinates doesn't help much, because at quick reading it's easy to miss them. So I don't want to add the flipping features back. Thank you anyway.

**Comment by davidair** [comment]
Good call, I've checked and actually all chess books do orient diagrams with white at the bottom, always. I kind of found seeing black at the bottom useful for my purposes (was printing thematic puzzles from lichess) but frankly, there is no point creating non-standard solutions. Note: it might be nice to have an indicator of whose turn it is (e.g. by drawing a black or white circle in the corner above the board). I tried doing it in Golang, realized the drawing was too low-level and ended up using Python/PIL :)
